Solutions/Corvus.Azure.Cosmos.Tenancy/Corvus/Azure/Cosmos/Tenancy/TenantCosmosContainerFactoryOptions.cs

```diff
@@ -0,0 +1,57 @@
+﻿// <copyright file="TenantCosmosContainerFactoryOptions.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Corvus.Azure.Cosmos.Tenancy
+{
+    using System;
+    using Corvus.Azure.Cosmos.Tenancy.Internal;
+
+    /// <summary>
+    /// Configuration settings for <see cref="TenantCosmosContainerFactory"/>.
+    /// </summary>
+    public class TenantCosmosContainerFactoryOptions
+    {
+        /// <summary>
+        /// Gets or sets the connection string to use when obtaining tokens as the service identity.
+        /// </summary>
+        /// <remarks>
+        /// <p>
+        /// In most circumstances, this will be null, indicating that the default behaviour should
+        /// be used. The default is that when running in an Azure App Service environment (e.g., a
+        /// Function), the Managed Identity will be used. When running locally during development,
+        /// it will look for a Visual Studio-supplied token file, and if there isn't one it then
+        /// tries asking the Azure CLI for a token. (These behaviours all come from the
+        /// <c>AzureServiceTokenProvider</c> - they are not specific to this library.)
+        /// </p>
+        /// <p>
+        /// If you want to be able to set this string via settings (e.g., in a local.settings.json
```

Comment:
The `local.settings.json` use case is important, but I wonder whether it's also good to point out that this is also suitable code for picking up settings passed in by Azure App Service. And we might actually want to show a couple of examples, because the code you show here works only when you have an `IConfiguration` in hand. That will be the case in an ASP.NET Core Web app, because the typical startup code for those involves building your config up and then defining your services.

But with an Azure Function they make it unnecessarily awkward to get hold of the configuration. The _only_ way to get it is from DI. This is frustrating because a) it means there's no way to obtain the configuration while DI setup is in progress, and b) it means you've got no choice but to have an `IConfiguration` in your DI services, something we prefer to avoid.

So in Azure Functions you end up doing something more like `services.AddSingleton(sp => new Something(sp.GetRequiredService<IConfiguration>().Get<SomethingOptions>());`

Actually, this has me wondering whether the `AddTenantCosmosContainerFactory` method shown here is actually problematic in an Azure Function, because it's written to presume that you are able to load your configuration ahead of setting up your DI container, which the Azure Functions SDK currently does not support.

Reply:
Have added extra method signatures, as per discussion and what was previously applied in Corvus.Leasing.